ethstore/src/accounts_dir/disk.rs

```diff
@@ -166,17 +167,17 @@ impl<T> DiskDirectory<T> where T: KeyFileManager {
 
 			// save the file
 			let mut file = fs::File::create(&keyfile_path)?;
-			if let Err(err) = self.key_manager.write(original_account, &mut file).map_err(|e| Error::Custom(format!("{:?}", e))) {
-				drop(file);
-				fs::remove_file(keyfile_path).expect("Expected to remove recently created file");
-				return Err(err);
-			}
+
+			// Write key content
+			self.key_manager.write(original_account, &mut file).map_err(|e| Error::Custom(format!("{:?}", e)))?;
```

Comment:
Wouldn't the simple `to_string()` do the trick here?

Reply:
It's a custom `Error` declared as enum. It only has `Debug` and `Display` implementations.